lib/db/derivedtables/led_segment_display.ts

```diff
@@ -0,0 +1,89 @@
+import type { DerivedTableSpec } from "./types"
+import type { KyselyDatabaseInstance } from "../kysely-types"
+import { extractMinQPrice } from "lib/util/extract-min-quantity-price"
+
+interface LEDSegmentDisplay {
+  lcsc: number
+  mfr: string
+  description: string
+  stock: number
+  price1: number | null
+  in_stock: boolean
+  package?: string
+  positions?: string
+  type?: string
+  size?: string
+  color?: string
+}
+
+export const ledSegmentDisplayTableSpec: DerivedTableSpec<LEDSegmentDisplay> = {
+  tableName: "led_segment_display",
+  extraColumns: [
+    { name: "package", type: "text" },
+    { name: "positions", type: "text" },
+    { name: "type", type: "text" },
+    { name: "size", type: "text" },
+    { name: "color", type: "text" },
+  ],
+
+  listCandidateComponents(db: KyselyDatabaseInstance) {
+    return db
+      .selectFrom("components")
+      .innerJoin("categories", "components.category_id", "categories.id")
+      .selectAll()
+      .where((eb) => eb("description", "like", "%LED Segment Display%"))
+  },
+
+  mapToTable(components) {
+    return components.map((c) => {
+      try {
+        const extraData = c.extra ? JSON.parse(c.extra) : {}
+        const attrs = extraData.attributes || {}
+
+        // Extract positions from description (e.g., "3 positions")
+        let positions = undefined
+        const posMatch = c.description.match(/(\d+)\s*positions?/)
```

Comment:
you need to match for capital P in positions too. Look at the page in your screenshot, see how it doesn't always work?